<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0" stdsetdef="1">
  <author></author>
  <comment></comment>
  <exportmacro></exportmacro>
  <class>KLSerialTerminalWidgetBase</class>
  <widget class="QWidget" name="KLSerialTerminalWidgetBase">
    <property name="geometry">
      <rect>
        <x>0</x>
        <y>0</y>
        <width>372</width>
        <height>467</height>
      </rect>
    </property>
    <property name="windowTitle">
      <string>Serial terminal</string>
    </property>
    <layout class="QGridLayout">
      <property name="margin">
        <number>4</number>
      </property>
      <item row="0" column="0">
        <widget class="QLabel" name="tlDevice">
          <property name="sizePolicy">
            <sizepolicy>
              <hsizetype>0</hsizetype>
              <vsizetype>5</vsizetype>
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
            </sizepolicy>
          </property>
          <property name="text">
            <string>Device:</string>
          </property>
          <property name="wordWrap">
            <bool>false</bool>
          </property>
        </widget>
      </item>
      <item row="0" column="3">
        <widget class="QLabel" name="tlBaud">
          <property name="sizePolicy">
            <sizepolicy>
              <hsizetype>0</hsizetype>
              <vsizetype>5</vsizetype>
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
            </sizepolicy>
          </property>
          <property name="text">
            <string>Baud rate:</string>
          </property>
          <property name="wordWrap">
            <bool>false</bool>
          </property>
        </widget>
      </item>
      <item row="0" column="1" rowspan="1" colspan="2">
        <widget class="QComboBox" name="cbDevice">
          <property name="editable">
            <bool>true</bool>
          </property>
        </widget>
      </item>
      <item row="0" column="4">
        <widget class="QComboBox" name="cbBaud"/>
      </item>
      <item row="0" column="5">
        <widget class="QToolButton" name="tbOpenClose">
          <property name="text">
            <string/>
          </property>
          <property name="toolTip" stdset="0">
            <string>Open/close the port</string>
          </property>
        </widget>
      </item>
      <item row="6" column="0" rowspan="1" colspan="6">
        <widget class="Line" name="line1_2">
          <property name="frameShape">
            <enum>QFrame::HLine</enum>
          </property>
          <property name="frameShadow">
            <enum>QFrame::Sunken</enum>
          </property>
        </widget>
      </item>
      <item row="3" column="0" rowspan="1" colspan="6">
        <widget class="Line" name="line1">
          <property name="frameShape">
            <enum>QFrame::HLine</enum>
          </property>
          <property name="frameShadow">
            <enum>QFrame::Sunken</enum>
          </property>
        </widget>
      </item>
      <item row="4" column="0" rowspan="1" colspan="6">
        <layout class="QHBoxLayout">
          <item>
            <widget class="QLabel" name="tlSendData">
              <property name="text">
                <string>Data to send (#xy where xy is in HEX, sends the value xy):</string>
              </property>
              <property name="alignment">
                <set>Qt::AlignVCenter</set>
              </property>
              <property name="wordWrap">
                <bool>true</bool>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QPushButton" name="pbSend">
              <property name="enabled">
                <bool>false</bool>
              </property>
              <property name="sizePolicy">
                <sizepolicy>
                  <hsizetype>0</hsizetype>
                  <vsizetype>0</vsizetype>
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                </sizepolicy>
              </property>
              <property name="text">
                <string>Se&amp;nd</string>
              </property>
              <property name="shortcut">
                <string>Alt+N</string>
              </property>
            </widget>
          </item>
        </layout>
      </item>
      <item row="7" column="0" rowspan="1" colspan="2">
        <layout class="QHBoxLayout">
          <item>
            <widget class="QLabel" name="tlReceiveData">
              <property name="text">
                <string>Data received:</string>
              </property>
              <property name="alignment">
                <set>Qt::AlignVCenter</set>
              </property>
              <property name="wordWrap">
                <bool>true</bool>
              </property>
            </widget>
          </item>
          <item>
            <widget class="QToolButton" name="tbClear">
              <property name="sizePolicy">
                <sizepolicy>
                  <hsizetype>4</hsizetype>
                  <vsizetype>4</vsizetype>
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                </sizepolicy>
              </property>
              <property name="text">
                <string/>
              </property>
              <property name="toolTip" stdset="0">
                <string>Clear the receiver buffer</string>
              </property>
            </widget>
          </item>
        </layout>
      </item>
      <item row="1" column="0" rowspan="1" colspan="6">
        <widget class="Q3ButtonGroup" name="bgFlowControl">
          <property name="title">
            <string>Flow control</string>
          </property>
          <layout class="QHBoxLayout">
            <property name="margin">
              <number>4</number>
            </property>
            <property name="spacing">
              <number>0</number>
            </property>
            <item>
              <widget class="QRadioButton" name="rbRTSCTS">
                <property name="text">
                  <string>&amp;RTS/CTS</string>
                </property>
                <property name="shortcut">
                  <string>Alt+R</string>
                </property>
                <property name="checked">
                  <bool>true</bool>
                </property>
              </widget>
            </item>
            <item>
              <widget class="QRadioButton" name="rbXONXOFF">
                <property name="text">
                  <string>XON/XOFF</string>
                </property>
                <property name="shortcut">
                  <string/>
                </property>
              </widget>
            </item>
            <item>
              <widget class="QRadioButton" name="rbNone">
                <property name="text">
                  <string>None</string>
                </property>
              </widget>
            </item>
          </layout>
        </widget>
      </item>
      <item row="2" column="0" rowspan="1" colspan="6">
        <widget class="QCheckBox" name="cbDisconnectWhileProgrammerActive">
          <property name="text">
            <string>Disconnect &amp;while programmer is active</string>
          </property>
          <property name="shortcut">
            <string>Alt+W</string>
          </property>
          <property name="checked">
            <bool>true</bool>
          </property>
        </widget>
      </item>
      <item row="5" column="0" rowspan="1" colspan="6">
        <widget class="Q3TextEdit" name="teSendData">
          <property name="sizePolicy">
            <sizepolicy>
              <hsizetype>7</hsizetype>
              <vsizetype>1</vsizetype>
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
            </sizepolicy>
          </property>
          <property name="font">
            <font>
              <family>Courier</family>
            </font>
          </property>
        </widget>
      </item>
      <item row="8" column="0" rowspan="1" colspan="2">
        <widget class="Q3ButtonGroup" name="bgShowAs">
          <property name="sizePolicy">
            <sizepolicy>
              <hsizetype>0</hsizetype>
              <vsizetype>5</vsizetype>
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
            </sizepolicy>
          </property>
          <property name="title">
            <string>Show as</string>
          </property>
          <layout class="QGridLayout">
            <item row="0" column="0">
              <widget class="QComboBox" name="cbShowAs">
                <item>
                  <property name="text">
                    <string>ASCII</string>
                  </property>
                </item>
                <item>
                  <property name="text">
                    <string>HEX</string>
                  </property>
                </item>
                <item>
                  <property name="text">
                    <string>Scope</string>
                  </property>
                </item>
                <item>
                  <property name="text">
                    <string>Histogram</string>
                  </property>
                </item>
              </widget>
            </item>
          </layout>
        </widget>
      </item>
      <item row="7" column="2" rowspan="2" colspan="4">
        <layout class="QVBoxLayout">
          <item>
            <widget class="Q3TextEdit" name="teReceiveData">
              <property name="sizePolicy">
                <sizepolicy>
                  <hsizetype>7</hsizetype>
                  <vsizetype>1</vsizetype>
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                </sizepolicy>
              </property>
              <property name="font">
                <font>
                  <family>Courier</family>
                </font>
              </property>
              <property name="readOnly">
                <bool>true</bool>
              </property>
              <property name="undoRedoEnabled">
                <bool>false</bool>
              </property>
            </widget>
          </item>
          <item>
            <widget class="Q3Frame" name="frmScopeHisto">
              <property name="frameShape">
                <enum>QFrame::StyledPanel</enum>
              </property>
              <property name="frameShadow">
                <enum>QFrame::Raised</enum>
              </property>
            </widget>
          </item>
        </layout>
      </item>
    </layout>
  </widget>
  <layoutdefault spacing="6" margin="11"/>
  <pixmapfunction>qPixmapFromMimeSource</pixmapfunction>
  <connections>
    <connection>
      <sender>tbOpenClose</sender>
      <signal>clicked()</signal>
      <receiver>KLSerialTerminalWidgetBase</receiver>
      <slot>slotOpenClose()</slot>
    </connection>
    <connection>
      <sender>pbSend</sender>
      <signal>clicked()</signal>
      <receiver>KLSerialTerminalWidgetBase</receiver>
      <slot>slotSend()</slot>
    </connection>
    <connection>
      <sender>tbClear</sender>
      <signal>clicked()</signal>
      <receiver>KLSerialTerminalWidgetBase</receiver>
      <slot>slotClearReceiver()</slot>
    </connection>
    <connection>
      <sender>cbShowAs</sender>
      <signal>activated(int)</signal>
      <receiver>KLSerialTerminalWidgetBase</receiver>
      <slot>slotShowAsChanged(int)</slot>
    </connection>
  </connections>
</ui>
